### PR TITLE
mon: osdmap's epoch should be more than 0

### PR DESCRIPTION
**Pull request by x11562**
Signed-off-by: Na Xie xie.na@h3c.com

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -3106,7 +3106,7 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
     int64_t epochnum;
     cmd_getval(g_ceph_context, cmdmap, "epoch", epochnum, (int64_t)0);
     epoch = epochnum;
-    if (!epoch)
+    if (epoch > 0)
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: does this change make any difference? or it is just cosmetic?

Comment (x11562) [inline src/mon/OSDMonitor.cc]:
```
ceph osd getmap 0
```
such as command, get the latest osdmap.
this change tells that 0 is invalid.

Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: this change breaks the behaviour of 
`ceph osd getmap`.

in this case, `epoch` is 0.

Comment (x11562) [inline src/mon/OSDMonitor.cc]: 0 is invalid

Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: the default value of `epoch` is "0" if it's not specified. we use it to tell if user wants to get the latest osdmap.

with your change, if user puts `ceph osd getmap`. he/she will get an error.
```
$ ./bin/ceph osd getmap
Error ENOENT: there is no map for epoch 0
```

Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: further more, you are overwriting the epoch passed by user with the latest one. this does not sound right.

Comment (x11562) [inline src/mon/OSDMonitor.cc]: if epoch is not specified, the latest epoch returned，rather than setted to 0, 
what do you think?

**Comment by dachary** [comment]
jenkins test this please (jenkins hung)

**Comment by tchaikov** [comment]
@jecluis agreed. but the change improves the readability, because the last param of `cmd_getval()` is the default value of the val of the command option, and we are using "0" as a flag actually. so i lean towards merging it after running it through rados qa.

**Comment by jecluis** [comment]
@tchaikov fair enough. thumbsup.png

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2016-11-25_08:09:33-rados-wip-kefu-testing2---basic-mira/

**Pull request merged**